packages/usa-input-mask/src/index.js

```diff
@@ -1,124 +1,1091 @@
   content.classList.add(MASK_CONTENT);
   content.setAttribute("aria-hidden", "true");
-  content.id = `${input.id}Mask`;
+  content.id = `${inputEl.id}Mask`;
   content.textContent = placeholder;
 
   shell.appendChild(content);
-  input.closest(CONTEXT).insertBefore(shell, input);
-  shell.appendChild(input);
+  inputEl.parentNode.insertBefore(shell, inputEl);
+  shell.appendChild(inputEl);
 };
 
-const setValueOfMask = (el) => {
-  const { value } = el;
-  const placeholderVal = `${el.dataset.placeholder.substr(value.length)}`;
+/**
+ * Returns an object containing the current status message and whether or not the
+ * character type is valid.
+ * @param {HTMLElement} inputEl - The input element.
+ * @param {number} keyCode - The key code of the key that was pressed.
+ * @param {string} key - The character that was entered.
+ * @param {number} curPos - The current position in the input field.
+ * @returns {{currentStatusMessage: string, invalidCharType: boolean}} An object containing the current status message and whether or not the character type is valid.
+ */
+const getMaskMessage = (inputEl, keyCode, key, curPos) => {
+  const isNumber = /^\d+$/.test(key);
+  const theCharType = !isNumber ? "number" : "letter";
 
-  const theIEl = document.createElement("i");
-  theIEl.textContent = value;
-  return [theIEl, placeholderVal];
+  const MASK = getMaskInfo(inputEl.id, "MASK", []);
+
+  if (isValidCharacter(keyCode, MASK[curPos])) {
+    return { currentStatusMessage: "", invalidCharType: false };
+  }
+
+  return {
+    currentStatusMessage: `A ${theCharType} ${DEFAULT_STATUS_LABEL}`,
+    invalidCharType: true,
+  };
 };
 
-const strippedValue = (isCharsetPresent, value) =>
-  isCharsetPresent ? value.replace(/\W/g, "") : value.replace(/\D/g, "");
+/**
+ * Updates the character count status for screen readers after a 200ms delay.
+ *
+ * @param {HTMLElement} msgEl - The screen reader status message element
+ * @param {string} statusMessage - A string of the current character status
+ */
+const srUpdateStatus = debounce((msgEl, statusMessage) => {
+  const srStatusMessage = msgEl;
+  srStatusMessage.textContent = statusMessage;
+}, 200);
 
-const isInteger = (value) => !Number.isNaN(parseInt(value, 10));
+/**
+ * Hides a status message by setting its text content and screen-reader-only status message element's text content to an empty string, and removing the `MESSAGE_INVALID_CLASS` class from the status message element.
+ *
+ * @param {HTMLElement} inputEl - The input element associated with the status message to be hidden.
+ */
+const hideMessage = (inputEl) => {
+  const { maskedEl } = getMaskedElements(inputEl, true);
+  const statusMessage = maskedEl.querySelector(STATUS_MESSAGE);
+  const srStatusMessage = maskedEl.querySelector(STATUS_MESSAGE_SR_ONLY);
 
-const isLetter = (value) => (value ? value.match(/[A-Z]/i) : false);
+  statusMessage.textContent = "";
+  srUpdateStatus(srStatusMessage, "");
```

Comment:
Check the character count error functionality... `usa-input-mask__message` holds that initial message, while `usa-input-mask__status` is what gets populated with the status message.

`MESSAGE` is used in the getMaskedElements function to confirm that the `usa-input-mask__message` element exists, otherwise it throws an error (just like character count).

`const messageEl = maskedEl.querySelector(MESSAGE);`

I'm working on getting the initial message to show again after valid or invalid inputs are entered.

Reply:
I addressed your initial comment in this section. The default message is now visible in all cases that an invalid input is not detected.

https://user-images.githubusercontent.com/19338309/215361701-c405bdf9-65e2-416b-8219-6b37ddaa4b3a.mov